lesson06.md

```diff
@@ -0,0 +1,427 @@
+
+We want to avoid rewriting nearly identical code.
+
+Functions have the option to accept **parameters** that let us do just that.
+
+---
+
+### Function parameters and arguments
+
+```js
+function sayHi(name) {
+  console.log("Hi " + name);
+}
+
+sayHi('Alan');
+sayHi('Ada');
+sayHi('Brendan');
+```
+
+- `name` is a function **parameter**
+- "Alan", "Ada", "Brendan" are **arguments**. In each function call they will be assigned to the `name` parameter
+
+---
+
+### Multiple parameters
+
+A function can have any number of parameters
+
+```js
+function sum(parameter1, parameter2, parameter3) {
+  console.log(parameter1 + parameter2 + parameter3);
+}
+
+introduce(1, 2, 3);
+```
+
+Each argument will be assigned to the corresponding parameter in order, from left to right
+
+---
+
+```js
+function greetFriend(greeting, name) {
+  console.log(greeting + " " + name + ", how are you today?");
+}
+
+greetFriend("Hi", "Alan");
+greetFriend("Good day", "Ada");
+```
+
+---
+
+## Returning values
+
+Typically, we use functions to make a calculation. We want to use the result of that calculation outside of the function.
+
+We can use the `return` keyword for that.
+
+---
+
+### Return: example
+
+```js
+function giveMe5() {
+  return 5;
+}
+
+let number = giveMe5();
+// number is 5
+```
+
+---
+
+### Quiz
+
+What is the value of the variable `result`?
+
+```js
+function add(a, b) {
+  return a + b;
+}
+
+let result = add(3, 5);
+```
+
+Result is 8
+<!-- .element: class="fragment" -->
+
+---
+
+### Returning values
+
+The `return` keyword exits the function immediately. Any code after the `return` statement will not be executed.
+
+```js
+function describeWeather(temperature) {
+  if (temperature > 25) {
+    return "It's hot";
+    // no further code will be executed
+  }
+  if (temperature < 5) {
+    return "It's cold";
+    // no further code will be executed
+  }
+  return "It's nice outside";
+  console.log("function finished"); // this will never be reached
+}
+```
+
+---
+
+### Common mistakes
+
+What is the value of the variable `number`?
+
+```js
+function giveMe5() {
+  console.log("5");
+}
+
+let number = giveMe5();
+console.log(number); // ???
+```
+
+`undefined`! This function does not have a return value!
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz
+
+What is the value of `result`?
+
+```js
+function giveMe5() {
+  return 5;
+}
+
+let result = giveMe5() * giveMe5();
+```
+
+Answer: 25
+<!-- .element: class="fragment" -->
+
+
+---
+
+### Task 2
+
+Write a `min` function that returns the smaller of the two numbers.
+
+```js
+min(1, 2);    // should return 1
+min(100, 99); // should return 99
+min(-10, 0);  // should return -10
+```
+
+---
+
+### Task 3
+
+Write a function that takes the name of a person, their age, and the language they speak, and returns a string that introduces this person.
+
+Example:
+
+Ada, 30, English &#8594; **"Hello! My name is Ada, I am 30 years old and I speak English."**
+
+---
+
+### Bonus Tasks
```

Comment:
I probably have a different understanding of what we mean with Homework - my understanding was that we would rename the last tasks of each lesson to be homeworks.
Do you not see any need for this label change?